src/program/program.dto.ts

```diff
@@ -0,0 +1,204 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { DisplayMode } from './program.type';
+
+export class ProgramResponseDTO {
+  @ApiProperty()
+  id: string;
+
+  @ApiProperty()
+  title: string;
+
+  @ApiProperty()
+  coverUrl: string;
+
+  @ApiProperty()
+  coverMobileUrl: string;
+
+  @ApiProperty()
+  coverThumbnailUrl: string;
+
+  @ApiProperty()
+  abstract: string;
+}
+
+class AudioDataDTO {
+  @ApiProperty()
+  name: string;
+
+  @ApiProperty()
+  size: number;
+
+  @ApiProperty()
+  type: string;
+
+  @ApiProperty()
+  lastModified: number;
+}
+
+class AudioDTO {
+  @ApiProperty({ type: AudioDataDTO })
+  data: AudioDataDTO;
+}
+
+class ProgramContentBodyDataDTO {
+  @ApiProperty()
+  data: Record<string, any>;
+
+  @ApiProperty()
+  description: string;
+
+  @ApiProperty()
+  id: string;
+
+  @ApiProperty()
+  type: string;
+}
+
+class AttachmentDto {
+  @ApiProperty()
+  id: string;
+
+  @ApiProperty()
+  data: any;
+
+  @ApiProperty()
+  type: string;
+
+  @ApiProperty()
+  target: string;
+
+  @ApiProperty()
+  createdAt: Date;
+
+  @ApiProperty()
+  updatedAt: Date;
+
+  @ApiProperty()
+  options: any | null;
+
+  @ApiProperty()
+  appId: string;
+
+  @ApiProperty()
+  isDeleted: boolean;
+
+  @ApiProperty()
+  contentType: string;
+
+  @ApiProperty()
+  name: string;
+
+  @ApiProperty()
+  size: string;
+
+  @ApiProperty()
+  authorId: string;
+
+  @ApiProperty()
+  thumbnailUrl: string;
+
+  @ApiProperty()
+  filename: string;
+
+  @ApiProperty()
+  duration: string;
+
+  @ApiProperty()
+  status: string;
+
+  @ApiProperty()
+  family: any | null;
+
+  @ApiProperty()
+  fileId: any | null;
+}
+
+class MetadataDTO {
+  private: boolean;
+  difficulty: number;
+  isCoverRequired: boolean;
+}
+
+export class ProgramContentResponseDTO {
+  @ApiProperty()
+  appid: string;
+
+  @ApiProperty()
+  id: string;
+
+  @ApiProperty()
+  title: string;
+
+  @ApiProperty()
+  abstract: string | null;
```

Comment:
word abstract in ts refers to a keyword , Is there any problem here ?
maybe can abstract_ or other word